bats/fb-katello-client-global-registration.bats

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bats
+# vim: ft=sh:sw=2:et
+
+set -o pipefail
+
+load os_helper
+load foreman_helper
+load fixtures/content
+
+@test "remove subscription manager" {
+  if [ -e "/etc/rhsm/ca/candlepin-local.pem" ]; then
+    rpm -e `rpm -qf /etc/rhsm/ca/candlepin-local.pem`
+  fi
+
+  cleanSubscriptionManager
+  tPackageRemove subscription-manager
+}
+
+@test "disable puppet agent to prevent checkin from registering host to another org" {
+  systemctl is-active puppet || skip "Puppet is not active"
+  systemctl stop puppet
+}
+
+@test "delete host if present" {
+  hammer host delete --name=$HOSTNAME || echo "Could not delete host"
+}
+
+@test "register with global registration with activation key" {
+  run yum erase -y 'katello-ca-consumer-*'
+  echo "rc=${status}"
+  echo "${output}"
+
+  registration_command=$(curl https://admin:changeme@$HOSTNAME/api/registration_commands -X POST -H 'Content-Type: application/json' -d '"activation_key":"${ACTIVATION_KEY}","organization":"${ORGANIZATION_LABEL}"' | ruby -e "require 'json'; puts JSON.load(ARGF.read)['registration_command']")
```

Comment:
This appears to not work as it generates the registration command with `Default_Organization` instead of `Test_Organization` not respecting passing the organization label? Is this a bug or am I setting this wrong? @ares @stejskalleos 

Additionally, I noticed that if I pass a parameter that registeration does not support there is not an error, the parameter is just ignored. That led to some debugging confusion

Reply:
For what it's worth, you shouldn't have to post using JSON. You can also shorten the Ruby line a bit.
```suggestion
  registration_command=$(curl https://admin:changeme@$HOSTNAME/api/registration_commands -X POST -d "activation_key=${ACTIVATION_KEY}" -d "organization=${ORGANIZATION_LABEL}" | ruby -rjson -e "puts JSON.load(ARGF.read)['registration_command']")
```

Reply:
I think you need to pass IDs, not names/labels for org?

https://github.com/theforeman/foreman/blob/5a66daa44921e63522bb90d01f8924b6cd4abcd5/app/controllers/concerns/foreman/controller/registration.rb#L17-L20

Reply:
> Additionally, I noticed that if I pass a parameter that registeration does not support there is not an error, the parameter is just ignored. That led to some debugging confusion

What if I told you, ALL of our API behaves like this, and I might have gray hair because of that?

Reply:
Yea... @ares was saddening me with this. Just makes debugging that much harder in this case to know what my client is doing.



> I think you need to pass IDs, not names/labels for org?

Yea... for this to work right now I will have to do that. I left it like to to first help the conversation about adding name/label support for organizations.

Reply:
I wonder if the user's default org works here. But I guess you want to make sure, user can use a specific org anyway... I can imagine adding another top-level parameter to specify org using the label, but that's not related to the registration.

Reply:
> I wonder if the user's default org works here. But I guess you want to make sure, user can use a specific org anyway... I can imagine adding another top-level parameter to specify org using the label, but that's not related to the registration.

What is the course of action here? I should file a bug?

Reply:
I would call it an RFE :-) I guess we shouldn't block the test on it. Note that if you insisht on org label and not the name, it must be added from Katello, since label is a katello thing. Perhaps it's time to move this extension to Foreman core where it belongs to?

Reply:
I can file an RFE for the label for Katello. Should *name* work though?

Reply:
> I can file an RFE for the label for Katello. Should _name_ work though?

Oh I guess not from what @evgeni linked. I'll file an issue for that too.

Reply:
> I would call it an RFE :-) I guess we shouldn't block the test on it. Note that if you insisht on org label and not the name, it must be added from Katello, since label is a katello thing. Perhaps it's time to move this extension to Foreman core where it belongs to?

By extension, you mean the concept of a label on organization to Foreman core?

Reply:
I filed the following for consideration:

 * https://projects.theforeman.org/issues/33227
 * https://projects.theforeman.org/issues/33228

Reply:
> For what it's worth, you shouldn't have to post using JSON. You can also shorten the Ruby line a bit.

The API only accepts JSON when using POST.